v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstopubsub/FailsafeModJsonToPubsubMessageTransformer.java

```diff
@@ -186,67 +220,78 @@ private PubsubMessage publishModJsonStringToPubsubMessage(String modJsonString)
-                                        break;
-                                    default:
-                                        break block;
-                                }
-                                changelogEntry.customEncode(encoder);
-                                encoder.flush();
-
-                                // Publish the encoded object as a Pub/Sub message.
-                                ByteString data = ByteString.copyFrom(byteStream.toByteArray());
-                                PubsubMessage message = PubsubMessage.newBuilder().setData(data).build();
-                                System.out.println("Publishing message: " + message);
-
-                                ApiFuture<String> future = publisher.publish(message);
-                                System.out.println("Published message ID: " + future.get());
-
-
-                            case "Protocol Buffer":
-                                ChangelogEntryProto changelogEntryProto = ChangelogEntryProto.newBuilder().build();
-                                switch(messageEncoding) {
-                                    case "BINARY":
-                                        break;
-
-                                    case "JSON":
-
-                                }
-                            case "JSON":
-                                byte[] encodedRecords = modJsonString.getBytes();
-
-                                com.google.pubsub.v1.PubsubMessage v1PubsubMessage =
-                                        com.google.pubsub.v1.PubsubMessage.newBuilder()
-                                                .setData(ByteString.copyFrom(encodedRecords))
-                                                .build();
-                                ApiFuture<String> messageIdFuture = publisher.publish(v1PubsubMessage);
-                                List<ApiFuture<String>> futures = new ArrayList();
-                                futures.add(messageIdFuture);
-                                ApiFutures.allAsList(futures).get();
-                                return v1PubsubMessage;
-                            default:
-                                final String errorMessage =
-                                        "Invalid output format:"
-                                                + messageFormat
-                                                + ". Supported output formats: JSON, AVRO";
-                                LOG.info(errorMessage);
-                                throw new IllegalArgumentException(errorMessage);
-                        }
-                    } else {
-
+                    ByteArrayOutputStream byteStream = new ByteArrayOutputStream();
+                    if (!topic.getSchemaSettings().getSchema().isEmpty()){
+                        messageFormat = topic.getSchemaSettings().getSchema();
```

Comment:
I am looking at this: https://screenshot.googleplex.com/4VJzLp4rDpXL2eU and it looks like the expected values for `messageFormat` are "AVRO" or "Protocol Buffer" or "JSON", but are you sure that these are the right values to expect? The doc says it's more like an identifier.